### PR TITLE
Use release build for OSS CI runner

### PR DESCRIPTION
**Pull request by kirklandsign**
Now we can enable release build.

### CODE REVIEW

File: .ci/scripts/gather_test_models.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def model_should_run_on_event(model: str, event: str) -> bool:
     We put higher priority and fast models to pull request and rest to push.
     """
     if event == "pull_request":
-        return model in ["add", "ic3", "mv2", "mv3", "resnet18", "vit"]
+        return True or model in ["add", "ic3", "mv2", "mv3", "resnet18", "vit"]
```
Comment (kirklandsign) [inline .ci/scripts/gather_test_models.py:75]: I will remove this once I get result.

Comment (kirklandsign) [reply .ci/scripts/gather_test_models.py:75]: https://hud.pytorch.org/pytorch/executorch/commit/1f5993a750d58d3de6102337c871c98718c25b9b

**Comment by kirklandsign** [comment]
Result with O2: https://hud.pytorch.org/pytorch/executorch/pull/1079?sha=f561b00ada5e6341a8770ac53c6a0b3f334c6d31

File: CMakeLists.txt [hunk 2]
```diff
@@ -83,10 +83,11 @@ if(NOT EXECUTORCH_ENABLE_PROGRAM_VERIFICATION)
...
 endif()
 
-# -Os: Optimize for size. -ffunction-sections -fdata-sections: breaks function
-# and data into sections so they can be properly gc'd. -s: strip symbols
+# -O2: Moderate opt. -ffunction-sections -fdata-sections: breaks function
+# and data into sections so they can be properly gc'd. -s: strip symbol.
+# -fno-exceptions -fno-rtti: disables exceptions and runtime type.
 set(CMAKE_CXX_FLAGS_RELEASE
-    "-Os -ffunction-sections -fdata-sections -fno-exceptions -fno-rtti")
+    "-O2 -ffunction-sections -fdata-sections -fno-exceptions -fno-rtti")
```
Comment (larryliu0820) [inline CMakeLists.txt:90]: I'm thinking maybe we need 2 modes, one for optimize performance one for optimize size. If we optimize for perf there will be stuff like loop unrolling which actually increase binary size.

**Pull request closed without merging**